src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -27,43 +29,43 @@ public ServerSentEventsTransport(ReadableChannel<Message> application, ILoggerFa
             context.Response.ContentType = "text/event-stream";
             context.Response.Headers["Cache-Control"] = "no-cache";
             context.Response.Headers["Content-Encoding"] = "identity";
+
             await context.Response.Body.FlushAsync();
 
+            var pipe = context.Response.Body.AsPipelineWriter();
+
             try
             {
                 while (await _application.WaitToReadAsync(token))
                 {
+                    var buffer = pipe.Alloc();
                     while (_application.TryRead(out var message))
                     {
-                        await Send(context, message);
+                        if (!ServerSentEventsMessageFormatter.TryFormatMessage(message, buffer.Memory.Span, out var written))
```

Comment:
These are basically internal (and could even be moved into an `Internal` namespace. In fact, I think I will do that). Also, they could end up unifying with `IStreamFormatters` eventually since they are attempting to achieve the same goal, writing an object to a stream.